src/main/java/com/jhipster/compositekey/service/mapper/BusinessBasicIndexMapper.java

```diff
@@ -16,17 +16,17 @@
 public interface BusinessBasicIndexMapper extends EntityMapper<BusinessBasicIndexDTO, BusinessBasicIndex> {
 
     @Mapping(source = "id.businessId", target = "businessId")
-    @Mapping(source = "business.name", target = "businessName")
     @Mapping(source = "id.basicIndexId", target = "basicIndexId")
-    @Mapping(source = "basicIndex.name", target = "basicIndexName")
     @Mapping(source = "id.year", target = "year")
+    @Mapping(source = "business.name", target = "businessName")
```

Comment:
Following along as I'm very interested.  I agree with what you're doing so far and it's very close to what I've implemented manually in my codebase.  For the mapper I've found that passing the id object in the DTO is cumbersome.  I use [@BeforeMapping](http://mapstruct.org/documentation/1.0/api/org/mapstruct/BeforeMapping.html) to create the Id object and set on the entity instead.